providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -0,0 +1,319 @@
+    const compatible = chains.filter((x) => this.isCompatibleChainId(x));
+    const chainIds = compatible.map((c) => this.parseChainId(c)).filter((c) => c !== this.chainId);
+    if (chainIds.length) {
+      this.chainId = chainIds[0];
+      this.events.emit("chainChanged", this.chainId);
+    }
+  }
+
+  private setChainId(chain: string) {
+    if (this.isCompatibleChainId(chain)) {
+      const chainId = this.parseChainId(chain);
+      this.chainId = chainId;
+      this.setHttpProvider(chainId);
+      this.events.emit("chainChanged", this.chainId);
+    }
+  }
+
+  private parseAccountId(account: string): { chainId: string; address: string } {
+    const [namespace, reference, address] = account.split(":");
+    const chainId = `${namespace}:${reference}`;
+    return { chainId, address };
+  }
+
+  private setAccounts(accounts: string[]) {
+    this.accounts = accounts
+      .filter((x) => this.parseChainId(this.parseAccountId(x).chainId) === this.chainId)
+      .map((x) => this.parseAccountId(x).address);
+    this.events.emit("accountsChanged", this.accounts);
+  }
+
+  private getRpcConfig(opts: EthereumProviderOptions): EthereumRpcConfig {
+    return {
+      chains: opts.chains.map((chain) => this.formatChainId(chain)) || [`${this.namespace}:1`],
+      methods: opts?.methods || signerMethods,
+      events: opts?.events || signerEvents,
+      rpcMap: opts?.rpcMap || this.buildRpcMap(opts.chains, opts.projectId),
+      showQrModal: opts?.showQrModal || true,
+      projectId: opts.projectId,
+      metadata: opts.metadata,
+    };
+  }
+
+  private buildRpcMap(chains: number[], projectId: string): EthereumRpcMap {
+    const map: EthereumRpcMap = {};
+    chains.forEach((chain) => {
+      map[chain] = this.getRpcUrl(chain, projectId);
+    });
+    return map;
+  }
+
+  private async initialize(opts: EthereumProviderOptions) {
+    this.rpc = this.getRpcConfig(opts);
+    this.chainId = getEthereumChainId(this.rpc.chains);
+    this.signer = await UniversalProvider.init({ projectId: this.rpc.projectId });
+    this.registerEventListeners();
+    if (this.rpc.showQrModal)
+      this.modal = new Web3Modal({
+        projectId: this.rpc.projectId,
+        standaloneChains: this.rpc.chains,
+      });
+  }
+
+  private loadConnectOpts(opts?: ConnectOps) {
+    if (!opts) return;
+    const { chains, rpcMap } = opts;
+    if (!isValidArray(chains) || !chains) return;
+    this.rpc.chains = this.rpc.chains.concat(chains.map((chain) => this.formatChainId(chain)));
+    // filter duplicate chains
+    this.rpc.chains = [...new Set(this.rpc.chains)];
+    chains.forEach((chain) => {
+      this.rpc.rpcMap[chain] = rpcMap?.[chain] || this.getRpcUrl(chain);
+    });
+  }
+
+  private getRpcUrl(chainId: number, projectId?: string): string {
+    const providedRpc = this.rpc.rpcMap?.[chainId];
+    return (
+      providedRpc ||
+      `https://rpc.walletconnect.com/v1/?chainId=eip155:${chainId}&projectId=${
```

Comment:
Not sure how this is currently set up in the other providers, but would be easier to create a constant (e.g. `RPC_URL`) that can easily be looked up and changed across providers in case the subdomain or path changes at a later point.